### PR TITLE
Add death animation

### PR DESCRIPTION
**Pull request by charasyn**
Upon taking fatal damage, the frog will spin around and play a sound
effect. After the sound effect ends, the player will be taken to the
title screen.

### CODE REVIEW

File: godotproject/Player.tscn [hunk 1]
```diff
@@ -16,19 +17,32 @@ random_pitch = 1.4
...
 tracks/0/enabled = true
 tracks/0/keys = {
-"times": PoolRealArray( 0.0021699, 0.104155, 0.20614, 0.305955 ),
-"transitions": PoolRealArray( 1, 1, 1, 1 ),
+"times": PoolRealArray( 0, 0.104155, 0.20614, 0.305955, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 1.1 ),
+"transitions": PoolRealArray( 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ),
 "update": 1,
-"values": [ 25, 26, 27, 28 ]
+"values": [ 25, 26, 27, 28, 25, 26, 27, 28, 25, 26, 27, 28 ]
+}
```
Comment (charasyn) [inline godotproject/Player.tscn:32]: We play the death animation of the frog rotating three times, to account for the length of the sound effect.

File: godotproject/Player.gd [hunk 2]
```diff
@@ -463,7 +470,22 @@ func takeDamage(damageTaken):
 		$InvulnerableTimer.start()
 
 func die():
-	# This will need to be modified to do the death animation once that is implemented.
+	# We've died!
+	dead = true
+	# Mark us as always executing
+	pause_mode = PAUSE_MODE_PROCESS
+	# Inhibit opening the menu
+	# TODO
```
Comment (fheysel) [inline godotproject/Player.gd:478]: Is this still a todo or can it be removed?

File: godotproject/Player.tscn [hunk 3]
```diff
@@ -16,19 +17,32 @@ random_pitch = 1.4
...
+tracks/1/loop_wrap = true
+tracks/1/imported = false
+tracks/1/enabled = true
+tracks/1/keys = {
+"times": PoolRealArray( 2.8 ),
+"transitions": PoolRealArray( 1 ),
+"values": [ {
+"args": [  ],
+"method": "death_animation_over"
+} ]
```
Comment (charasyn) [inline godotproject/Player.tscn:45]: This animation track calls the "death_animation_over" function in the player after 2.8 seconds (about the length needed to play the sound effect)

**Comment by fheysel** [comment]
looks good to me, make sure youve rebased off main and made sure everything still looks good

**Review by charasyn — COMMENTED** [review]
Some notes

**Review by fheysel — APPROVED** [review]

File: godotproject/Player.gd (hunk 2 above)
Comment (charasyn) [reply godotproject/Player.gd:478]: Oops

**Pull request merged**